homeassistant/components/upnp/__init__.py

```diff
@@ -211,6 +220,8 @@ def _substitute_hass_ports(ports, hass_port=None):
 
     # remove sensors
     _LOGGER.debug("Deleting sensors")
-    dispatcher.async_dispatcher_send(hass, SIGNAL_REMOVE_SENSOR, device)
+    hass.async_create_task(
```

Comment:
We should await the unload and return the result of the unload. That's a boolean.